quarkus/service/src/main/java/org/apache/polaris/service/quarkus/config/QuarkusProducers.java

```diff
@@ -261,4 +266,13 @@ public ActiveRolesProvider activeRolesProvider(
   public void closeTaskExecutor(@Disposes @Identifier("task-executor") ManagedExecutor executor) {
     executor.close();
   }
+
+  @Produces
+  public List<DataSource> dataSources(@All List<InstanceHandle<DataSource>> dataSources) {
+    List<DataSource> dataSourceList = new ArrayList<>();
+    for (InstanceHandle<DataSource> handle : dataSources) {
+      dataSourceList.add(handle.get());
+    }
+    return dataSourceList;
```

Comment:
With the fixed data source configurations, the [bootstrap tool ](https://polaris.apache.org/in-dev/unreleased/admin-tool/#bootstrapping-realms-and-principal-credentials) won't work with JDBC, as it requires the data source to be pre-configured. With that, adding a new account level tenant will require to stop the Polaris server, to add new data source and to restart the server. This makes database-base multi-tenancy impossible. We need to figure out a way to dynamically add data sources. Can we file an issue and track that effort?

Reply:
Here you go : https://github.com/apache/polaris/issues/1463